packages/ui-kit/src/components/multi-path-block.tsx

```diff
@@ -0,0 +1,212 @@
+        }
+        cursor: pointer;
+      }
+    `,
+];
+
+type TTabHeaderProps = {
+  label: string;
+  isActive: boolean;
+  onClick: (e: React.MouseEvent<HTMLElement>) => void;
+};
+
+const TabHeader = (props: TTabHeaderProps) => {
+  return (
+    <span
+      role="tab"
+      aria-selected={props.isActive}
+      key={props.label}
+      onClick={props.onClick}
+      css={getLinkStyles(props.isActive)}
+    >
+      <Text.Headline as="h3" truncate={true}>
+        {props.label}
+      </Text.Headline>
+    </span>
+  );
+};
+
+const SelectorsContainer = styled.div`
+  border-bottom: 1px solid ${designSystem.colors.light.borderPrimary};
+  padding-top: 8px;
+  padding-left: 16px;
+`;
+
+const ComponentWrapper = styled.div`
+  border: 2px solid ${designSystem.colors.light.borderPrimary};
+  border-radius: ${designSystem.tokens.borderRadiusForMultiPath};
+`;
+
+const PathsContainer = styled.div`
+  padding: 10px;
+`;
+
+const MultiPathBlock = (props: MultiPathBlockProps) => {
+  const labelSyncItems = extractLabelSyncPair(props.children);
+  const [selected, setSelected] = React.useState<LabelSyncPair>(
+    labelSyncItems[0]
+  );
+  const [activePathIndex, setActivePathIndex] = useState<number>(0);
+
+  const { selectedPath, updateSelectedPath } = useSelectedPath();
+
+  useEffect(() => {
+    if (selectedPath) {
+      const matchedSyncItem = labelSyncItems.find(
+        (item) => item.syncWith?.toLowerCase() === selectedPath // case insensitive match
+      );
+      if (matchedSyncItem) {
+        setSelected(matchedSyncItem);
+      }
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [selectedPath]);
+
+  useEffect(() => {
+    let activeIndex = Array.isArray(props.children)
+      ? props.children.findIndex(
+          (child) => child.props.label === selected.label
+        )
+      : -1;
+    setActivePathIndex(activeIndex !== -1 ? activeIndex : 0);
+  }, [props.children, selected]);
+
+  const onTabHeaderClick =
+    (e: React.MouseEvent<HTMLElement>) => (labelSyncItem: LabelSyncPair) => {
+      e.preventDefault();
+      e.stopPropagation();
+      if (labelSyncItem.label === selected.label) {
+        return;
+      }
+      setSelected(labelSyncItem);
+      if (labelSyncItem.syncWith) {
+        updateSelectedPath(labelSyncItem.syncWith);
+      } // if no syncWith, do nothing
+    };
+
+  return (
+    <ComponentWrapper>
+      <SelectorsContainer role="tablist">
+        <Spacings.Inline alignItems="flex-end">
+          {labelSyncItems.map((labelSyncItem) => (
+            <TabHeader
+              key={labelSyncItem.label}
+              label={labelSyncItem.label}
+              isActive={labelSyncItem.label === selected.label}
+              onClick={(e) => onTabHeaderClick(e)(labelSyncItem)}
+            />
+          ))}
+        </Spacings.Inline>
+      </SelectorsContainer>
+      <PathsContainer>
+        {Array.isArray(props.children) ? (
```

Comment:
What do you think about extracting this child/children component to a small component so we reduce the comments and logic in the markup. It might be helpful so we dont have comments in markup. 
```
const TabContentChildren = ...

LOC 189
<TabContentChildren />

```

Reply:
Yes, that makes the code more readable too